src/network.rs

```diff
@@ -223,7 +225,14 @@ impl Network {
                         *value = new_value;
                     }
 
-                    stack.push(*weight * new_value);
+                    if j {
+                        // when j flag is set,
+                        // do not include weight of link as jump forward has a different weight
+                        stack.push(new_value);
+                    } else {
+                        // otherwise use regular weight of connection in stack
+                        stack.push(*weight * new_value);
+                    }
```

Comment:
This should only apply to the final neuron, otherwise sub-neurons won't apply their weight.

Reply:
Yes you are right, my test case did not have a subnetwork of depth > 1 so I did not notice!